test/verify/check-s4u-ssh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import parent
+
+from testlib import *
+
+@skipImage("No realmd available", "fedora-coreos")
+@skipImage("freeipa not currently in testing", "debian-testing")
+class TestS4USsh(MachineCase):
+    provision = {
+        "0": {"address": "10.111.113.1/20", "dns": "10.111.112.100"},
+        "1": {"address": "10.111.113.2/20", "dns": "10.111.112.100"},
+        "services": {"image": "services", "memory_mb": 2048}
+    }
+
+    def setUp(self):
+        super().setUp()
+        self.machines['services'].execute("/run-freeipa")
+
+    def testBasic(self):
+        client_machine = self.machine
+        sshd_machine = self.machines["1"]
+        ipa_machine = self.machines["services"]
+
+        # set hostname for readable logs
+        client_machine.execute("hostnamectl set-hostname sshclient.cockpit.lan")
+        sshd_machine.execute("hostnamectl set-hostname sshserver.cockpit.lan")
+        self.login_and_go("/system")
+
+        # join both machines
+        wait(lambda: client_machine.execute("nslookup -type=SRV _ldap._tcp.cockpit.lan"))
+        wait(lambda: sshd_machine.execute("nslookup -type=SRV _ldap._tcp.cockpit.lan"))
+        wait(lambda: client_machine.execute("echo foobarfoo | realm join -U admin cockpit.lan"))
+        wait(lambda: sshd_machine.execute("echo foobarfoo | realm join -U admin cockpit.lan"))
+
+        # configure ipa
+        ipa_machine.execute("docker exec freeipa bash -c 'echo foobarfoo | kinit admin@COCKPIT.LAN'")
+        # add user to impersonate
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa user-add --first=user --last=user user'")
+        # ipa_machine.execute("echo foobarfoo | ipa user-mod user --password")
+        # add service which will be allowed to delegate creds
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa service-add cockpitclient/sshclient.cockpit.lan@COCKPIT.LAN --ok-as-delegate=true --ok-to-auth-as-delegate=true'")
+        # Allow restrieval of service/host keytabs by admin
```

Comment:
re"t"rieval. That looks like some weird kind of "unbreak my setup" step -- these commands are already run as user "admin", thus it has the power to grant that privilege to itself. I suppose the "by admin" is not the important part here, but the "by the involved services"?

What does the  `--user admin` option actually mean here? That a service like `cockpitclient/sshclient.cockpit.lan` can retrieve the keytab *of* the "admin" user? I. e. that's the moral equivalent of setuid root?

Reply:
It means the admin is allowed to retrieve the keytab for that service, so we can do it later in line.